src/test/java/com/gravity9/mongocse/ChangeStreamTest.java

```diff
@@ -283,6 +283,59 @@ void givenConfigurationWithUserDefinedKey_eachListenerShouldReceiveSeparateEvent
         assertEquals(2, events2.get(0).getFullDocument().getInteger("testValue"));
     }
 
+    @Test
+    void givenConfigurationWithMatchStageWithSingleCondition_shouldAcceptOnlyMatchingEvents() throws InterruptedException {
+        var match = Filters.gt("fullDocument.testValue", 0);
+        var config = buildMongoConfig(match);
+
+        MongoCseManager manager = new MongoCseManager(config);
+        TestChangeStreamListener listener = new TestChangeStreamListener();
+        manager.registerListenerToAllPartitions(listener);
+        manager.start();
+
+        insertDocumentsToAllPartitions();
+
+        // Wait for CDC event to arrive
+        Thread.sleep(500);
+
+        assertEquals(2, listener.getEvents().size());
+    }
+
+    @Test
+    void givenConfigurationWithMatchStageWithMultipleConditions_shouldAcceptOnlyMatchingEvents() throws InterruptedException {
```

Comment:
Please also a add a couple of test cases where filters work together with the basic key filtering. Also a test that checks if match filtering works in the partitioned environment (I'd recommend to use a custom field for that, not _id to make sure the tests are consistent).

Reply:
sure, done.